apps/glusterfs/brick_allocate.go

```diff
@@ -26,6 +27,7 @@ func NewBrickSet(s int) *BrickSet {
 }
 
 func (bs *BrickSet) Add(b *BrickEntry) {
+	godbc.Require(!bs.Full())
```

Comment:
What about adding the same check for `DeviceSet.Add()`?

Reply:
Of course. I'm just being a blockhead.